pay-api/src/pay_api/services/eft_short_names.py

```diff
@@ -592,11 +590,12 @@ def get_search_query(cls, search_criteria: EFTShortnamesSearch, is_count: bool =
 
         # Join payment information if this is NOT the count query
         if not is_count:
+            statement_summary_query = cls.get_statement_summary_query().subquery()
             query = cls.add_payment_account_name_columns(query)
             query = (query.add_columns(
                 statement_summary_query.c.total_owing,
                 statement_summary_query.c.latest_statement_id
-            ).outerjoin(
+            ).join(
```

Comment:
Make sure to test and confirm cases where an account has no statements or empty statements will still appear. This join could eliminate new accounts that have no statements yet from appearing.